### PR TITLE
Add bilateral filter

### PR DESCRIPTION
**Pull request by praveennadiminti**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
https://github.com/TheAlgorithms/Python/blob/master/Travis_CI_tests_are_failing.md
-->
https://travis-ci.com/TheAlgorithms/Python/builds/150259743#L779

File: digital_image_processing/filters/bilateral_filter.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+Output:
+    img:A 2d zero padded image with values in between 0 and 1
+'''
+
+import cv2
+import numpy as np
+import math
+import sys
+
+def vec_gaussian(img, var):
```
[the remark this answers is not in the document]
Comment (praveennadiminti) [reply digital_image_processing/filters/bilateral_filter.py]: added type hint

File: digital_image_processing/filters/bilateral_filter.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    # For applying gaussian function for each element in matrix.
+    sigma = math.sqrt(var)
+    cons = 1/(sigma*math.sqrt(2*math.pi))
+    fImg = cons*np.exp(-((img/sigma)**2)*0.5)
+    return fImg
+
+def getSlice(img,x,y, N):
```
[the remark this answers is not in the document]
Comment (praveennadiminti) [reply digital_image_processing/filters/bilateral_filter.py]: added type hint

File: digital_image_processing/filters/bilateral_filter.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    return img[x-N//2:x+N//2+1,y-N//2:y+N//2+1]
+
+def getGaussKernel(N, varS):
```
[the remark this answers is not in the document]
Comment (praveennadiminti) [reply digital_image_processing/filters/bilateral_filter.py]: added type hint

File: digital_image_processing/filters/bilateral_filter.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    # Creates a gaussian kernel of given dimension.
+    arr = np.zeros((N,N))
+    for i in range(0,N):
+        for j in range(0,N):
+            arr[i,j] = math.sqrt(abs(i-N//2)**2 + abs(j-N//2)**2)
+    arr = vec_gaussian(arr,varS)
+    return arr
+
+def bilateral_filter(img, varS, varI, N):
```
[the remark this answers is not in the document]
Comment (praveennadiminti) [reply digital_image_processing/filters/bilateral_filter.py]: added type hint

**Comment by praveennadiminti** [comment]
What do you think about the names kernalSize for N, intensityVariance for varI and spatialVariance for varS?

**Comment by cclauss** [comment]
Python variable names should be in `snake_case` as discussed in CONTRIBUTING.md.
> kernal_size, intensity_variance, spatial_variance

**Comment by praveennadiminti** [comment]
done.

File: digital_image_processing/filters/bilateral_filter.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+def getGaussKernel(kernel_size: int, spatial_variance: float) -> np.ndarray:
+    # Creates a gaussian kernel of given dimension.
+    arr = np.zeros((kernel_size, kernel_size))
+    for i in range(0, kernel_size):
+        for j in range(0, kernel_size):
+            arr[i, j] = math.sqrt(
+                abs(i - kernel_size // 2) ** 2 + abs(j - kernel_size // 2) ** 2
+            )
+    arr = vec_gaussian(arr, spatial_variance)
+    return arr
```
Comment (TheSuperNoob) [inline digital_image_processing/filters/bilateral_filter.py]: Should change these method names to be `snake_case` also

Comment (itsvinayak) [reply digital_image_processing/filters/bilateral_filter.py]: changed the style to snake_case

**Pull request merged**